limitador/src/storage/redis/counters_cache.rs

```diff
@@ -308,9 +323,23 @@ impl CountersCacheBuilder {
         self
     }
 
+    fn eviction_listener(
+        _key: Arc<Counter>,
+        value: Arc<CachedCounterValue>,
+        _removal_cause: RemovalCause,
+    ) {
+        gauge!("cache_size").decrement(1);
+        if let Ok(writes) = value.pending_writes() {
```

Comment:
You can't call that `pending_writes`... while the value got evicted from the cache, it's still in the queue! But now, when it'll get consumed, it won't have pending writes anymore... You can "just" call `no_pending_writes` on it. if it had none, then you can ignore this eviction, _but_ if we start evicting entries with pending writes, it probably means you need to review your cache size